doc/script_development.md

```diff
@@ -43,7 +43,9 @@ except for a few special ones:
 | `type` | a class, which is parse and validate the value given by the user |
 | `type_params` | optional parameters for the type class |
 | `depends_on` | makes this argument depend on the presence of another one |
-| `positional` | indicates whether the argument is a `--keyword` one (default) or positional. Must not be supplied together with `required'. |
+| `positional` | indicates whether the argument is a `--keyword` one (default) or positional. Must not be supplied together with `required`. |
+| `prompt` | prompt to ask the user for input, if the argument is not supplied. Works only if the argument is not `required`. |
```

Comment:
What happens if a dev supplies both a `prompt` and `required: true`? Is there an explicit error message in this case?

Reply:
Kinda… If you set `required` you __always__ have to provide the argument __on the command line__. It's okay, to set both, no harm done. But you never even get to the point where you will be prompted. 

If you don't set `required` but set `prompt`, you have to provide something too, but you can provide it to the prompt if it's missing on the command line. And if you don't set `required` but set `prompt` and `prompt_options['empty']`, you can even skip the prompt.

Reply:
No idea for a good way to reword this right now. But yes, should make it more clear, that it's okay but might not be what one wants.

Reply:
Maybe something like: 
>  prompt to ask the user for input, if the argument is not supplied. If the argument is `required`, it hat to be set on the command line though.